src/plugins/data_source_management/public/types.ts

```diff
@@ -140,3 +140,8 @@ export {
   SigV4Content,
   DataSourceAttributes,
 } from '../../data_source/common/data_sources';
+
+// flint datasources types
+export type DatasourceType = 'S3GLUE' | 'PROMETHEUS';
+
+export type DatasourceStatus = 'ACTIVE' | 'DISABLED';
```

Comment:
are these the only status? and i don't know if these status are unique to datasources